scripts/announcements.R

```diff
@@ -0,0 +1,192 @@
+
+# Installation and Loading Packages ---------------------------------------
+# Running these lines will install the necessary packages.
+
+if (!require(devtools)) install.packages("devtools")
+devtools::install_dev_deps()
+library(tidyverse)
+library(lubridate)
+library(glue)
+library(assertr)
+
+# Importing and Filtering the Event Data ----------------------------------
+
+session_details <- read_csv(here::here("_data", "events.csv"), comment = "#") %>%
+    arrange(date) %>%
+    # drop sessions that are not set (NA in date)
+    filter(!is.na(date))
+
+# Find any existing posts, take the date, and filter out those sessions from the
+# session_details dataframe.
+keep_only_new_sessions <- function() {
+    existing_post_dates <- fs::dir_ls(here::here("_posts"), regexp = ".md$|.markdown$") %>%
+        str_extract("[0-9]{4}-[0-9]{2}-[0-9]{2}")
+
+    session_details %>%
+        filter(!as.character(date) %in% existing_post_dates)
+}
+
+new_sessions <- keep_only_new_sessions()
+
+# Create a GitHub Issue of the session ------------------------------------
+
+post_gh_issue <- function(title, body, labels) {
+    # Will need to set up a GitHub PAT via (I think) the function
+    # devtools::github_pat() in the console.
+    devtools:::rule("Posting GitHub Issues")
+    cat("Posting `", title, "`\n\n")
+    if (!devtools:::yesno("Are you sure you want to post this event as an Issue?")) {
+        gh::gh(
+            "POST /repos/:owner/:repo/issues",
+            owner = "uoftcoders",
+            repo = "Events",
+            title = title,
+            body = body,
+            labels = array(c(labels))
+        )
+        usethis:::done("Event posted as an Issue to UofTCoders/Events.")
+        return(invisible())
+    } else {
+        message("Event not posted to Issue.")
+    }
+}
+
+# Format as eg August 23
+day_month <- function(.date, add_name = TRUE) {
+    date_format <- "%B %e" # as August 23
+    if (add_name) date_format <- "%A, %B %e" # as Monday, August 23
+    trimws(format(as.Date(.date), format = date_format))
+}
+
+gh_issue_info <- function(.data) {
+    content <- .data %>%
+        mutate(needs_packages = ifelse(
+            !is.na(packages),
+            str_c(
+                "Please also install these packages: ",
+                str_replace_all(packages, " ", ", "), "."
+            ),
+            ""
+        )) %>%
+        mutate_at(vars(start_time, end_time), funs(strftime(., format = "%H:%M"))) %>%
+        glue_data(
+            "
+            {description}
+
+            - **Where**: MADLab at [Gerstein Science Information Centre](https://goo.gl/maps/2916Y54jQkx)
+            - **When**: {day_month(date)}, from {start_time}-{end_time} pm
+            - **Instructor**: TBA
+            - **Skill level**: {skill_level}
+
+            *Installation instructions*: You will need to install the appropriate programs. See the {program_language} section of the [installation instructions page](?link?). {needs_packages}
```

Comment:
I need the link for the installation instructions, if we have it.

Reply:
do you mean this: https://github.com/UofTCoders/studyGroup/blob/gh-pages/lessons/r-python-conda-setup.md

Reply:
Thanks. Edited that and added to this PR.